sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/IndexPolicyCompareService.java

```diff
@@ -22,12 +22,13 @@ public class IndexPolicyCompareService {
      * @param newPolicy New policy
      * @return Whether the policy needs updating.
      */
-    public static boolean policyNeedsUpdate(IndexingPolicy existingPolicy, IndexingPolicy newPolicy) {
-        return !hasSameIncludedPaths(existingPolicy.getIncludedPaths(), newPolicy.getIncludedPaths())
+    public static boolean policyNeedsUpdate(IndexingPolicy existingPolicy, IndexingPolicy newPolicy, CosmosEntityInformation<?, ?> metadata) {
+        return ((!hasSameIncludedPaths(existingPolicy.getIncludedPaths(), newPolicy.getIncludedPaths())
             || !hasSameExcludedPaths(existingPolicy.getExcludedPaths(), newPolicy.getExcludedPaths())
             || !existingPolicy.getCompositeIndexes().equals(newPolicy.getCompositeIndexes())
             || !existingPolicy.getIndexingMode().equals(newPolicy.getIndexingMode())
-            || !existingPolicy.isAutomatic().equals(newPolicy.isAutomatic());
+            || !existingPolicy.isAutomatic().equals(newPolicy.isAutomatic()))
+            && metadata.getIndexingPolicyOverwritePolicy(newPolicy.getClass()));
```

Comment:
putting this check here with AND condition might be too confusing for readability. 

Worth checking this condition at the top and skipping everything else if the user has not set it to true.

Reply:
Done.